src/Features/VisualBasic/Portable/CodeRefactorings/SyncNamespace/VisualBasicSyncNamespaceService.vb

```diff
@@ -0,0 +1,79 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Collections.Immutable
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.CodeRefactorings.SyncNamespace
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.SyncNamespace
+    <ExportLanguageService(GetType(ISyncNamespaceService), LanguageNames.VisualBasic), [Shared]>
+    Friend Class VisualBasicSyncNamespaceService
+        Inherits AbstractSyncNamespaceService(Of NamespaceStatementSyntax, CompilationUnitSyntax)
+
+        Public Overrides Function TryGetReplacementReferenceSyntax(reference As SyntaxNode, newNamespaceParts As ImmutableArray(Of String), ByRef old As SyntaxNode, ByRef [new] As SyntaxNode) As Boolean
+            Dim nameRef = TryCast(reference, SimpleNameSyntax)
+            If nameRef IsNot Nothing Then
+                Dim outerMostNode = GetQualifiedNameSyntax(nameRef)
+                old = outerMostNode
+
+                If outerMostNode Is nameRef Or newNamespaceParts.IsDefaultOrEmpty Then
+                    [new] = outerMostNode
+                Else
+                    If newNamespaceParts.Length = 1 And newNamespaceParts(0).Length = 0 Then
+                        [new] = nameRef
+                    Else
+                        Dim qualifiedNamespaceName = CreateNameSyntax(newNamespaceParts, newNamespaceParts.Length - 1)
+                        [new] = SyntaxFactory.QualifiedName(qualifiedNamespaceName, nameRef)
+                    End If
+                    [new] = [new].WithTriviaFrom(outerMostNode)
+                End If
+                Return True
+            Else
+                old = Nothing
+                [new] = Nothing
+                Return False
+            End If
```

Comment:
how is thsi code different than the C# version?

Reply:
same comment applies.  These seems super similar, and they use syntax-facts a bunch.  Can we not pull this up into a common location?

Reply:
One of the issues is how alias qualifier is handled. I will give it a try.